test/integration/spec/remotetracks.js

```diff
@@ -0,0 +1,136 @@
+  this.timeout(60000);
+
+  describe('.setPriority', () => {
+    let thisRoom; let thoseRooms;
+    let aliceLocal; let bobLocal;
+    let aliceRemote; let bobRemote;
+    let aliceTracks; let bobTracks;
+    let aliceRemoteVideoTrack; let bobRemoteVideoTrack;
+
+    beforeEach(async () => {
+      const dataTrack = new LocalDataTrack();
+      [, thisRoom, thoseRooms] = await setup({
+        testOptions: {
+          bandwidthProfile: {
+            video: { maxTracks: 1 }
+          },
+          tracks: [dataTrack]
+        },
+        otherOptions: { tracks: [dataTrack] },
+        nTracks: 0
+      });
+
+      [aliceTracks, bobTracks] = await Promise.all([1, 2, 3].map(async () => [
+        createSyntheticAudioStreamTrack() || await createLocalAudioTrack({ fake: true }),
+        await createLocalVideoTrack(smallVideoConstraints),
+      ]));
+
+      [aliceLocal, bobLocal] = thoseRooms.map(room => room.localParticipant);
+      [aliceRemote, bobRemote] = [thisRoom.participants.get(aliceLocal.sid), thisRoom.participants.get(bobLocal.sid)];
+
+      // Alice publishes her tracks at low priority
+      // Bob publishes his tracks at medium priority
+      await Promise.all([
+        ...aliceTracks.map(track => aliceLocal.publishTrack(track, { priority: PRIORITY_LOW })),
+        ...bobTracks.map(track => bobLocal.publishTrack(track, { priority: PRIORITY_STANDARD })),
+        tracksSubscribed(aliceRemote, 3),
+        tracksSubscribed(bobRemote, 3)
+      ]);
+
+      [aliceRemoteVideoTrack, bobRemoteVideoTrack] = [aliceRemote, bobRemote].map(({ videoTracks }) => {
+        return [...videoTracks.values()][0].track;
+      });
+    });
+
+    afterEach(async () => {
+      [thisRoom, ...thoseRooms].forEach(room => room && room.disconnect());
+      if (thisRoom) {
+        await completeRoom(thisRoom.sid);
+      }
+    });
+
+    ['audio', 'video', 'data'].forEach(kind => {
+      [null, ...Object.values(trackPriority)].forEach(priority => {
+        it(`can setpriority ${priority} on ${kind} track`, () => {
+          const tracks = getTracksOfKind(aliceRemote, kind);
+          assert(tracks.length > 0, 'no tracks found of kind ' + kind);
+          tracks[0].setPriority(priority);
+          assert.equal(tracks[0].priority, priority);
+        });
+      });
+    });
+
+    [undefined, 'foo', 42].forEach(priority => {
+      it('throws for an invalid priority value: ' + priority, () => {
+        let errorWasThrown = false;
+        try {
+          aliceRemoteVideoTrack.setPriority(priority);
+        } catch (error) {
+          assert(error instanceof RangeError);
+          assert(/priority must be one of/.test(error.message));
+          errorWasThrown = true;
+        }
+        assert.equal(errorWasThrown, true, 'was expectiing an error to be thrown, but it was not');
+      });
+    });
+
+    if (defaults.topology !== 'peer-to-peer') {
+      it('subscriber can update track\'s effective priority', async () => {
```

Comment:
@makarandp0 Can you make sure you have tests for 3 scenarios:
1. Upgrading the effective RemoteVideoTrack priority.
2. Downgrading the effective RemoteVideoTrack priority.
3. Reverting to the publisher priority by passing `null`. (You can disable this test with a NOTE pointing to the VMS bug).

Reply:
I have a test for #1 - but #2 is no-op. As I understand subscriber can not lower effctive  priority of the track - They can only increase the effective priority.

Reply:
@makarandp0 According to David (who implemented subscriber priority), what you said only applies to Dominant Speaker tracks. Downgrading should work with `RemoteTrack.setPriority`. You can confirm with him if you want.

Reply:
okay, cool, I will add a test to downgrade priority and see it passes